pandas/core/reshape/melt.py

```diff
@@ -40,6 +41,14 @@ def melt(
     else:
         cols = list(frame.columns)
 
+    if value_name in frame.columns:
+        warnings.warn(
+            'The value column name "%s" conflicts with an existing '
+            "column in the dataframe." % (value_name),
```

Comment:
Generally we want to give users insight as to what they need to do before the deprecated behavior gets changed - can you try to clarify the message a bit?